core/src/main/scala/org/apache/spark/sql/TiExtensions.scala

```diff
@@ -33,6 +34,7 @@ class TiExtensions extends (SparkSessionExtensions => Unit) {
 
     e.injectResolutionRule(new TiAuthRuleFactory(getOrCreateTiContext))
     e.injectPlannerStrategy(new TiStrategyFactory(getOrCreateTiContext))
+    e.injectCheckRule(TelemetryRule)
```

Comment:
if you inject checkRule. The telemetry will do at the end of the analyzer. which means you can't execute telemetry once the analyzer fail such as `table or view not found` error.
if that's not your intention, maybe you should inject other stages

Reply:
Telemetry should not send messages in every SQL analysis. When TelemetryRule is built-in spark context, the message will be sent in TelemetryRule's construct function.